### PR TITLE
[14.0][ADD] epa_lab_sample_management: add new module

### PR DESCRIPTION
**Pull request by kaynnan**
cc @marcelsavegnago @Escodoo/consultores @Escodoo/desenvolvedores 

Task SO62-217

Refatoração do módulo Registros de Amostras desenvolvido pelo @DigoRameh por meio da [PR 52](https://github.com/Escodoo/epa-addons/pull/52)

### CODE REVIEW

**Review by WesleyOliveira98 — APPROVED** [review]
Laboratory notebook inside `res.partner`
image

Sample Form
image

Sample Tree
image

LGTM

File: epa_lab_sample_management/views/lab_sample.xml [hunk 1]
```diff
@@ -69,9 +84,10 @@
         <field name="model">lab.sample</field>
         <field name="arch" type="xml">
             <tree>
+                <field name="name" />
                 <field name="project_id" />
-                <field name="lab_id" />
-                <field name="matrix_id" />
+                <field name="laboratory" />
```
Comment (kaynnan) [inline epa_lab_sample_management/views/lab_sample.xml]: @WesleyOliveira98  acredito que podemos manter o padrão partner_id, por se referenciar ao `res.partner`, ai na String colocamos como Laboratory, seguindo o padrão do Odoo referente a nomenclaturas com id

Comment (WesleyOliveira98) [reply epa_lab_sample_management/views/lab_sample.xml]: É que já temos um campo `partner_id` que é o Cliente (Customer)

Comment (kaynnan) [reply epa_lab_sample_management/views/lab_sample.xml]: Putz, agora que fui ver hehe

Comment (WesleyOliveira98) [reply epa_lab_sample_management/views/lab_sample.xml]: Alterado para `laboratory_id`

**Comment by WesleyOliveira98** [comment]
Create a Matrix
image

Create a Parameter Group
image

Create a Parameter
image

Create a Sample Registration
image

The sample name follows the logic of the old system
image

On creation, all the required fields are logged into chatter
image

- When changing the company, the currency is filled with the company's currency
- When changing the project, the customer is filled with the project customer
- When changing parameter groups, the parameter list will get the parameters from the selected groups

**Comment by WesleyOliveira98** [comment]
Create a Matrix
image

Create a Parameter Group
image

Create a Parameter
image

Create a Sample Registration
image

The sample name follows the logic of the old system
image

On creation, all the required fields are logged into chatter
image

Server action for open the replicate wizard
image

Replicate Sample Wizard
image

- The `currency_id` field is related with the company's `currency_id`
- The customer (`partner_id`) is related with the analytic account `partner_id`
- When select parameter groups, the parameter list will get the parameters from the selected groups

**Comment by WesleyOliveira98** [comment]
- Add relation with purchase order
- The billing field is now related to the purchase order `amount_total`
- Add translations for pt_BR

**Comment by kaynnan** [comment]
ping @marcelsavegnago

**Comment by marcelsavegnago** [comment]
@WesleyOliveira98 @kaynnan We have already deployed this to staging, okay? When it is validated, please inform us so that we can merge this PR.

**Comment by kaynnan** [comment]
@WesleyOliveira98 Please confirm this issue, I don't remember if the customer had already given the ok.

**Comment by marcelsavegnago** [comment]
We deployed the module last week in the environment.

**Comment by WesleyOliveira98** [comment]
After validation, the client request some improves today, i'll work on that this week

**Comment by WesleyOliveira98** [comment]
Commited changes:

- Permission groups for users and administrators
- Record rules to restrict user write and unlink permissions.
- Depth is now a Char field
- Depth is included in the name when filled
- The sequential number in the name is generated by a sequence and computed upon sample creation
- Status is no longer computed; buttons are added to change the sample state
- Collection Date is now a Datetime field
- Default filters for samples created by the user and samples the user is following
- Most fields can only be filled or changed when in draft status

**Comment by WesleyOliveira98** [comment]
The client has validated, with the new changes, ready for merging @marcelsavegnago

**Comment by WesleyOliveira98** [comment]
force push to add translations, ready for merge

**Pull request merged**